sgpc3/sgpc3.c

```diff
@@ -713,6 +711,112 @@ s16 sgp_set_iaq_baseline(u16 baseline) {
 }
 
 
+/**
+ * sgp_set_absolute_humidity() - set the absolute humidity for compensation
+ *
+ * The absolute humidity must be provided in in mg/m^3 and the value must
+ * be between 0 and 256000 mg/m^3.
+ * If the absolute humidity is set to zero, humidity compensation is disabled
+ *
+ * @absolute_humidity:   u32 absolute humidity in mg/m^3
+ *
+ * Return:      STATUS_OK on success, else STATUS_FAIL
+ */
+s16 sgp_set_absolute_humidity(u32 absolute_humidity) {
+    const struct sgp_profile *profile;
+    u16 ah_scaled;
+    const u16 BUF_SIZE = SGP_COMMAND_LEN + SGP_WORD_LEN + CRC8_LEN;
+    u8 buf[BUF_SIZE];
+
+    profile = sgp_get_profile_by_number(PROFILE_NUMBER_SET_ABSOLUTE_HUMIDITY);
+
+    if (profile == NULL)
+        return STATUS_FAIL;
+
+    if (absolute_humidity > 256000)
+        return STATUS_FAIL;
+
+    /* ah_scaled = (absolute_humidity / 1000) * 256 */
+    ah_scaled = (u16)(((u64)absolute_humidity * 256 * 16777) >> 24);
+
+    sgp_fill_cmd_send_buf(buf, &profile->command, &ah_scaled,
+                          sizeof(ah_scaled) / SGP_WORD_LEN);
+
+    if (sensirion_i2c_write(SGP_I2C_ADDRESS, buf, BUF_SIZE) != 0)
+        return STATUS_FAIL;
+
+    return STATUS_OK;
+}
+
+
+/**
+ * sgp_get_iaq_factory_baseline() - read out the factory baseline from the chip
+ *                                  for the currently set power mode
+ *
+ * The IAQ factory baseline can be retrieved for a faster sensor startup.
+ * See application notes for further documentation.
+ *
+ * This functions returns STATUS_FAIL if the factory baseline value is not
+ * available.
+ *
+ * @baseline:   Pointer to raw u16 where to store the factory baseline
+ *              If the method returns STATUS_FAIL, the baseline value must be
+ *              discarded and must not be passed to sgp_set_iaq_baseline().
+ *
+ * Return:      STATUS_OK on success, else STATUS_FAIL
+ */
+s16 sgp_get_iaq_factory_baseline(u16 *baseline) {
+    if ((client_data.info.feature_set_version & 0x00FF) < 5)
```

Comment:
Thanks. Using the return value of `sgp_get_profile_by_number` is the wrong way since, in the worst case, a different profile could exist on the same index (but breaking changes like that require a different major feature set version).